feat(multimodal): Add customizable background color for RGBA to RGB conversion

ahengljh opened the pull request:
# Essential Elements of an Effective PR Description Checklist

  - [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
  - [x] The test plan, such as providing test command.
  - [x] The test results, such as pasting the results comparison before and after, or e2e results
  - [x] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.


  ## Purpose

  This PR implements customizable background color support for RGBA to RGB image conversion, addressing the TODO comment in `vllm/multimodal/image.py:26`.

  Previously, when converting RGBA images to RGB, the background was hard-coded to white (255, 255, 255). This limited flexibility for users who need:
  - Dark theme integration (black backgrounds)
  - Brand-specific color schemes
  - Model-specific requirements for background colors

## Test Plan

  Run the new test to verify the feature works correctly:
  ```bash
  pytest tests/multimodal/test_image.py::test_rgba_to_rgb_custom_background -v

  For manual testing via CLI:
  # Test with black background
  python -m vllm.entrypoints.openai.api_server \
      --model <your-vision-model> \
      --media-io-kwargs '{"image": {"rgba_background_color": [0, 0, 0]}}'

  # Test with custom color (e.g., gray)
  python -m vllm.entrypoints.openai.api_server \
      --model <your-vision-model> \
      --media-io-kwargs '{"image": {"rgba_background_color": [128, 128, 128]}}'
```

## Test Result

tests/multimodal/test_image.py::test_rgba_to_rgb_custom_background PASSED

## (Optional) Documentation Update

 Created docs/custom_rgba_background.md with:
  - Feature overview and use cases
  - Python API usage examples
  - CLI usage examples with --media-io-kwargs
  - Technical details about backward compatibility

  Added comprehensive docstring to ImageMediaIO class with usage examples.

ahengljh committed:
Add customiable background color

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

DarkLight1337 commented on docs/custom_rgba_background.md:
```diff
@@ -0,0 +1,75 @@
+# Custom RGBA Background Color Feature
```
Let's move this into a separate section under Multimodal Inputs

ahengljh committed:
Move document to proper place as comments required.

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

ahengljh committed:
Pre-submit changes

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

ahengljh committed:
Add input validation for rgba_background_color parameter

- Validate that rgba_background_color is a tuple/list of 3 integers
- Ensure all values are in the range [0, 255]
- Add comprehensive tests for validation edge cases
- Prevent potential crashes from invalid user input

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

ahengljh committed:
Pre-submit changes again

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

DarkLight1337 commented on docs/features/multimodal_inputs.md:
```diff
@@ -655,3 +655,49 @@ The following example demonstrates how to pass image embeddings to the OpenAI se
+
+When loading RGBA images (images with transparency), vLLM converts them to RGB format. By default, transparent pixels are replaced with white background. You can customize this background color using the `rgba_background_color` parameter in `media_io_kwargs`.
+
+#### Usage
```
I think this should belong under Image Inputs. Also the Usage subheader is redundant since we have code admonitions already

ahengljh committed:
Move documentation changes to proper location according to the comment

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

ahengljh commented on docs/features/multimodal_inputs.md:
````diff
@@ -172,6 +172,50 @@ Multi-image input can be extended to perform video captioning. We show this with
+    )
+    ```
+
+??? code "Online Inference"
````
Oh I see, you meant that we should only keep only the offline inference example in the Image Inputs section, and move the online inference example to the "Online Serving" section. Is that right?

> There is a separate section for online inference, put this code there

DarkLight1337 replied:
Yes that's correct

ahengljh committed:
Reorganize RGBA background color documentation and some minor changes for pre-submit checking.

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

DarkLight1337 commented on tests/multimodal/test_image.py:
```diff
@@ -35,3 +35,123 @@ def test_rgba_to_rgb():
+
+def test_rgba_background_color_validation():
+    """Test that invalid rgba_background_color values are properly rejected."""
+    import pytest
```
No need to lazy import pytest

ahengljh committed:
As comment said, use tmp_path and import pytest at the header part

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

DarkLight1337 reviewed (approved):
Thanks for implementing this!
